src/wp-admin/includes/ajax-actions.php

```diff
@@ -162,6 +162,8 @@ function wp_ajax_ajax_tag_search() {
 		)
 	);
 
+	$results = apply_filters( 'wp_after_tag_search', $results );
```

Comment:
1. Filters should have docblocks accompanying them. See line 137 for an example.

2. We should consider a guard so that we only use the filtered results if the returned value is an array.

3. This PR should also have PHPUnit tests that make use of this filter.

Reply:
> We should consider a guard so that we only use the filtered results if the returned value is an array.

I don't understand the benefit of doing this.

[get_terms](https://developer.wordpress.org/reference/functions/get_terms/) returns
```
WP_Term[] | int[] | string[] | string | WP_Error
````

Safeguarding the results and making sure only an `array` is passed, may be confusing and unnecessary?

Reply:
> This PR should also have PHPUnit tests that make use of this filter.

Unfortunately, I am not familiar with testing. I'm guessing you are expecting another function [here](https://github.com/WordPress/wordpress-develop/blob/e8435082d6df01068a45acc2f86658e8ac105d6f/tests/phpunit/tests/ajax/TagSearch.php)?

Reply:
TL;DR - Let's skip the guard and we'll pick it up when [this ticket](https://core.trac.wordpress.org/ticket/51525) is resolved.

get_terms will return:
- `string` for count queries, but this isn't a count query.
- `WP_Error` if the taxonomy doesn't exist. However, we already check this on line 114.
- That leaves only `WP_Term[] | int[] | string[]` as possible returns values from `get_terms`. Note that we are running `implode()` on the results, which requires an array as the second argument.

However, we are introducing a filter.

A callback may use functions that will often return `array`, but may also return `false`. Yes, callbacks should validate, but edge cases can be missed. It would be prudent to ensure that, after filtering, `$results` is still an array, in order, to help the function maintain some of its stability.

However, let's skip this bit, as we'll be doing a full review of Core when [this ticket](https://core.trac.wordpress.org/ticket/51525) is resolved.

Reply:
Regarding tests:

Take a look [here](https://github.com/WordPress/wordpress-develop/blob/b1a1bdd88c5b816175da9ac2da9789827eba3298/tests/phpunit/tests/ajax/TagSearch.php#L43). This test method will be very similar to one you'll be adding for this filter.

Suggested steps:

- Copy this test method to the bottom of the file.
- Add an appropriate docblock for the method, which should include:
  - Description.
  - `@ticket 55606`.
  - `@covers ::wp_ajax_ajax_tag_search`
- Give the method an appropriate name, such as `test_should_apply_wp_after_tag_search_filters()`.
- Before `// Make the request`, use the new filter. You can do this using a static closure, like so:

```php
add_filter(
    'wp_after_tag_search'
    static function( $results, $tax, $s ) {
        $results = array( 'wp_after_tag_search was applied' );
    }
);
```

You won't need to remove the filter afterwards - the test suite handles the clean up of filters between tests.

Feel free to ping me for another review/if you have any questions. :)

Reply:
Thank you!

I've attempted to create [a new test](https://github.com/grandeljay/wordpress-develop/blob/55e7d94bfd3f68522504d9b6b27915e940f34dda/tests/phpunit/tests/ajax/TagSearch.php#L159-L191) but it keeps failing. As far as I can tell it fails at [/wp-includes/class-wp-hook.php:307](https://github.com/WordPress/wordpress-develop/blob/7819ca4e3e3d2c0b3dbfec0ec83f52c304ef69bd/src/wp-includes/class-wp-hook.php#L307).

The error
```php
 1) Tests_Ajax_TagSearch::test_after_tag_search_filter
ArgumentCountError: Too few arguments to function Tests_Ajax_TagSearch::{closure}(), 1 passed in /var/www/src/wp-includes/class-wp-hook.php on line 309 and exactly 3 expected

/var/www/tests/phpunit/tests/ajax/TagSearch.php:177
/var/www/src/wp-includes/class-wp-hook.php:309
/var/www/src/wp-includes/plugin.php:191
/var/www/src/wp-admin/includes/ajax-actions.php:174
/var/www/src/wp-includes/class-wp-hook.php:307
/var/www/src/wp-includes/class-wp-hook.php:331
/var/www/src/wp-includes/plugin.php:476
/var/www/tests/phpunit/includes/testcase-ajax.php:265
/var/www/tests/phpunit/tests/ajax/TagSearch.php:184
phpvfscomposer:///var/www/vendor/phpunit/phpunit/phpunit:97
```

What am I missing?